python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -650,6 +672,48 @@ def _get_entities_from_payload(self, data):
         else:
             raise RuntimeError("Unable to determine an entity from data: %s" % data)
 
+    def _get_entity_type_whitelist(self, project_id):
+        """
+        Gets a set of entity types that are supported by the browser
+        integration. This set is built from a list of constant entity
+        types, plus all entity types that a PublishedFile entity is
+        allowed to link to, per the current site's schema.
+
+        :param int project_id: The associated project entity id. The
+            schema is queried by project, as project-level masking of the
+            PublishedFile entity's linkable types is possible. If no project
+            is given, the site-level schema is queried instead.
+
+        :returns: A set of string entity types.
+        :rtype: set
+        """
+        if project_id is None:
+            logger.debug("Project id is None, looking up site schema.")
+            project_entity = None
+        else:
+            project_entity = dict(type="Project", id=project_id)
+            logger.debug("Looking up schema for project %s", project_entity)
+
+        cache_is_initialized = self.ENTITY_TYPE_WHITELIST in self._cache
+        project_in_cache = project_id in self._cache.get(self.ENTITY_TYPE_WHITELIST, dict())
+
+        if not cache_is_initialized or not project_in_cache:
+            type_whitelist = constants.BASE_ENTITY_TYPE_WHITELIST
```

Comment:
It's still required. The PublishedFile's entity link types does not typically include the types in the constant, but we always want to support those, regardless.